faunadb-java/src/main/java/com/faunadb/client/types/time/HighPrecisionTime.java

```diff
@@ -59,52 +64,58 @@ private static int toInt(String value) {
    * @param microsToAdd microseconds to add to the initial timestamp
    * @param nanosToAdd  nanoseconds to add to the initial timestamp
    */
-  public HighPrecisionTime(Instant initialTime, int microsToAdd, int nanosToAdd) {
+  public static HighPrecisionTime fromInstant(Instant initialTime, long microsToAdd, long nanosToAdd) {
```

Comment:
Should we ignore micros at all? Because we can't have `fromInstant(Instant inst, long nanos)` and `fromInstant(Instant inst, long micros)` since they have the same signature.